src/elements/MetricsSelector/MetricsSelector.js

```diff
@@ -42,10 +42,20 @@ import { ReactComponent as ResultIcon } from 'igz-controls/images/circled-r.svg'
 
 import './metricsSelector.scss'
 
-const MetricsSelector = ({ maxSelectionNumber, metrics, name, onSelect, preselectedMetrics }) => {
+const MetricsSelector = ({
+  maxSelectionNumber,
+  metrics,
+  modelEndpointUid,
+  name,
+  onSelect,
+  preselectedMetrics
+}) => {
   const [nameFilter, setNameFilter] = useState('')
   const [isOpen, setIsOpen] = useState(false)
+  const [preAppliedMetrics, setPreAppliedMetricsRef] = useState([])
+  const appliedMetricsRef = useRef([])
```

Comment:
I don't see why we need to have `appliedMetrics` as a ref and not as a state.

Reply:
change to state